src/collection-view.js

```diff
@@ -158,6 +158,62 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
+  _addChildModels(models) {
+    let buffer = document.createDocumentFragment();
+    let triggerOnChildren = [];
+
+    this._startUpdating();
+    const addedViews = _.reduce(models, (adding, model, index) =>{
+      let view = this.children.findByModel(model);
+
+      if (!view) {
+        this._destroyEmptyView();
+
+        view = this._addChild(model, index);
+        adding.push(view);
+        triggerOnChildren.push(view);
+        buffer.appendChild(view.el);
+      } else if (buffer.children.length) {
+        if (this.sort) {
+          view._index = index;
+        }
+
+        _.each(triggerOnChildren, child => {
```

Comment:
> Is it possible to end up in a situation where this.children.findByModel(model) returns a view but the _.reduce isn't finished yet?

I could safe guard using something like `this._processingView = {};` and set/check against that until this is done, then reset to an empty object when done.